include/networkit/graph/BFS.hpp

```diff
@@ -0,0 +1,108 @@
+#ifndef GRAPH_BFS_H
+#define GRAPH_BFS_H
+
+#include <stack>
+
+#include <networkit/graph/Graph.hpp>
+
+namespace NetworKit {
+
+namespace GraphTraversal {
+
+/**
+ * Calls the given BFS handle with distance parameter
+ */
+template <class F>
+auto callBFSHandle(F &f, node u, count dist) -> decltype(f(u, dist)) {
```

Comment:
I just implemented a new `GraphTools` algortihm that needs access to the same `edgeLambda` dispatcher functionality used by Graph itself. Hence I factored out all those dispatcher functions into an own header. I suggest that we keep it here as is, but refactor it as soon as this PR is merged.